Add note about changing saved datasets (#11)
inrupt/docs-client-lib-js: source/tutorial/read-write-data.rst

If the given URL already contains a dataset, that dataset is replaced
with the updated dataset.

.. admonition:: After saving a SolidDataset
   :class: important

   A SolidDataset keeps track of what data has changed compared to
   the data in the Pod, and will refuse to save data if the data in

>    the Pod is not what it expects.

Reviewer: Thanks @Vinnl  -- Quick question -  is it that the SolidData set will refuse to save data or the saveSolidDatasetAt will refuse to save data?

```suggestion
A SolidDataset keeps track of data changes compared to the data in the Pod. When using ``saveSolidDatasetAt`` to write to an existing SolidDataset, the SolidDataset to save must differ from the one in the Pod.  This means, you cannot make multiple calls to ``saveSolidDatasetAt`` with the same SolidDataset. You must modify the SolidDataset first. For instance,  you cannot call ``saveSolidDatasetAt`` on its returned SolidDataset (e.g., ``savedSolidDataset`` in the example above) without first modifying it.
```

Author: Technically, it's the server that refuses to save the data. The `SolidDataset` itself contains the list of what data should be deleted and what should be added, and then the server will verify that the data that it says should be deleted is actually present before deleting it. (So the issue is not that there are no modifications, but that the modifications do not match what's at the server. For example, I can send the modification "change my name from `Vincent` to `Vinnie`", but when my name isn't currently `Vincent` (for example, because I've already saved the change before, and so the name on my Pod is already `Vinnie`), then the server will refuse that modification.

Reviewer: Oh - then people can save the returned SolidDataset object again w/o modification (as long as no one else has change the dataset separately)?

Author: That is correct (although it would of course be slightly pointless).

Reviewer: That is, call saveSolidDatasetAt won't fail, I mean (unless someone has changed the Dataset separately), since no changes to apply

Reviewer: Oh, then is it just 

```suggestion
A SolidDataset keeps track of the data changes compared to the data in the Pod; i.e., the changelog tracks both the old value and new values of the property being modified. If the old version does not correspond to the value currently in the Pod, the save operation will error. 
```

Reviewer: Oh - and I just recognized that I'm doing the suggestion syntax incorrectly - drat.

> .. admonition:: After saving a SolidDataset
>    :class: important
> 
>    A SolidDataset keeps track of what data has changed compared to
>    the data in the Pod, and will refuse to save data if the data in
>    the Pod is not what it expects.
>    This means you cannot call ``saveSolidDatasetAt`` on the same
>    SolidDataset instance. Instead, make sure to do further data
>    manipulation on the return value of ``saveSolidDatasetAt``,
>    i.e. ``savedSolidDataset`` in the example above.

Reviewer: ```suggestion
A SolidDataset keeps track of the data changes compared to the data in the Pod; i.e., the changelog tracks both the old value and new values of the property being modified. If the old version does not correspond to the value currently in the Pod, the save operation will error. 
```

Reviewer: Oh -- I just looked at the source.  My understanding was off a bit.  I see that for updates, we track it as a remove + add in the change log.  Will tweak.

Reviewer: If your okay with the suggestion, we can merge it in and then I can add a fuller example to illustrate the behavior.
